app_dart/lib/src/request_handlers/update_agent_health.dart

```diff
@@ -45,22 +49,55 @@ class UpdateAgentHealth extends ApiRequestHandler<UpdateAgentHealthResponse> {
         throw BadRequestException('Invalid agent ID: $agentId');
       },
     );
+    final TabledataResourceApi tabledataResourceApi = await config.createTabledataResourceApi();
```

Comment:
This line only makes sense with _insertBigQuery(), therefore it should be encapsulate within that function. Otherwise it's unexpected to see a big query api occurs among data store updating logic.

Reply:
Put it inside.

Reply:
Removed this line.

Reply:
moved.